addon/services/lazy-loader.js

```diff
@@ -55,7 +55,45 @@ export default Ember.Service.extend({
       return Ember.RSVP.resolve();
     }
 
-    return loadAssets(bundle.urls).then(()=>this.markBundleAsLoaded(bundle.name));
+    var urlsToLoad = this.getAssetUrlsForBundle(bundle);
+    return loadAssets(urlsToLoad).then(()=>this.markBundleAsLoaded(bundle.name));
+  },
+
+  /*
+
+    urlsToLoad: map of url -> true -- need O(1) set/get for adding items to the collection. passed recursively because bottom-up generation is O(n)
+    bundlesHandled: map of bundle -> true -- same as above, need O(1) collection management. used for cycle avoidance.
+
+  */
+  getAssetUrlsForBundle(bundle) {
+    var bundlesHandled = {};
```

Comment:
I think all of this would change based on what we talk in person, however if this was the way I was implementing it I would add a default parameter to the function instead of having the consumer do it. This greatly simplifies its signature from `_getAssetUrlsForBundle(bundle, urlsToLoad, bundlesHandled)` to `_getAssetUrlsForBundle(bundle):[]`